### PR TITLE
hotfix/CP-4583 app version filter activity null

### PR DESCRIPTION
**Pull request by jprajapati1996**

### CODE REVIEW

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 1]
```diff
@@ -460,7 +460,8 @@ public SessionListener initSessionListener() {
                     this.initFeatures();
                 }
 
-                if (this.appBannerModule != null) {
+                Logger.d("initSession", "appBannerModule" + this.getCurrentActivity());
```
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:463]: remove log

**Review by cpKevinrodriguesnobre — CHANGES_REQUESTED** [review]
we still need a log in case we early exit because of activity == null

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java (hunk 1 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java]: `    if (this.appBannerModule != null && this.getCurrentActivity() != null) {
                    this.appBannerModule.initSession(channelId);
                } else {
                    Logger.e("LOG_TAG", "activity null: " + getCurrentActivity());
                }`

Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java]: is it okay?

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 2]
```diff
@@ -470,6 +470,8 @@ public SessionListener initSessionListener() {
 
                 if (this.appBannerModule != null && this.getCurrentActivity() != null) {
                     this.appBannerModule.initSession(channelId);
+                } else {
```
Comment (cpKevinrodriguesnobre) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:473]: change else to else if(this.getCurrentActivity() == null)
or just if(this.getCurrentActivity() == null)

Comment (cpKevinrodriguesnobre) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java:473]: and change
"activity null: " + getCurrentActivity()
to 
"getCurrentActivity() is null"

**Review by cpKevinrodriguesnobre — CHANGES_REQUESTED** [review]

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java (hunk 2 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java:473]:
```
                if (this.appBannerModule != null && this.getCurrentActivity() != null) {
                    this.appBannerModule.initSession(channelId);
                } else if(this.getCurrentActivity() == null){
                    Logger.e("LOG_TAG", "getCurrentActivity() is null");
                }

```

**Review by cpKevinrodriguesnobre — APPROVED** [review]
not sure if we need the changes in AndroidManifest.xml though

File: cleverpush-example/src/main/AndroidManifest.xml [hunk 3]
```diff
@@ -8,32 +8,34 @@
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
 
     <application
-        android:name="com.example.cleverpush.App"
+        android:name=".App"
```
Comment (gbhrdt) [inline cleverpush-example/src/main/AndroidManifest.xml:11]: Revert all changes in AndroidManifest

File: cleverpush-example/src/main/java/com/example/cleverpush/MainActivity.java [hunk 4]
```diff
@@ -2,6 +2,7 @@
 
 import static com.cleverpush.CleverPushHttpClient.BASE_URL;
 
+import android.content.Intent;
```
Comment (gbhrdt) [inline cleverpush-example/src/main/java/com/example/cleverpush/MainActivity.java:5]: revert

File: cleverpush-example/src/main/java/com/example/cleverpush/MainActivity2.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
+package com.example.cleverpush;
```
Comment (gbhrdt) [inline cleverpush-example/src/main/java/com/example/cleverpush/MainActivity2.java:1]: revert

File: cleverpush-example/src/main/res/layout/activity_main2.xml [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (gbhrdt) [inline cleverpush-example/src/main/res/layout/activity_main2.xml:1]: revert

File: cleverpush/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -8,15 +8,22 @@
     <application>
         <activity
             android:name="com.cleverpush.NotificationOpenedActivity"
```
Comment (gbhrdt) [inline cleverpush/src/main/AndroidManifest.xml:10]: revert all changes here

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java [hunk 8]
```diff
@@ -468,8 +468,10 @@ public SessionListener initSessionListener() {
                     this.initFeatures();
                 }
 
-                if (this.appBannerModule != null) {
+                if (this.appBannerModule != null && this.getCurrentActivity() != null) {
                     this.appBannerModule.initSession(channelId);
+                } else if(this.getCurrentActivity() == null){
```
Comment (gbhrdt) [inline cleverpush/src/main/java/com/cleverpush/CleverPush.java:473]: code style! space after if missing

**Review by gbhrdt — CHANGES_REQUESTED** [review]

File: cleverpush-example/src/main/AndroidManifest.xml (hunk 3 above)
Comment (jprajapati1996) [reply cleverpush-example/src/main/AndroidManifest.xml]: Done

File: cleverpush-example/src/main/java/com/example/cleverpush/MainActivity.java (hunk 4 above)
Comment (jprajapati1996) [reply cleverpush-example/src/main/java/com/example/cleverpush/MainActivity.java]: Done

File: cleverpush-example/src/main/java/com/example/cleverpush/MainActivity2.java (hunk 5 above)
Comment (jprajapati1996) [reply cleverpush-example/src/main/java/com/example/cleverpush/MainActivity2.java]: Done

File: cleverpush-example/src/main/res/layout/activity_main2.xml (hunk 6 above)
Comment (jprajapati1996) [reply cleverpush-example/src/main/res/layout/activity_main2.xml]: Done

File: cleverpush/src/main/AndroidManifest.xml (hunk 7 above)
Comment (jprajapati1996) [reply cleverpush/src/main/AndroidManifest.xml:10]: Done

File: cleverpush/src/main/java/com/cleverpush/CleverPush.java (hunk 8 above)
Comment (jprajapati1996) [reply cleverpush/src/main/java/com/cleverpush/CleverPush.java]: Added

**Review by cpKevinrodriguesnobre — APPROVED** [review]

**Pull request merged**